sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2357,7 +2404,8 @@ pi_result cuda_piQueueFlush(pi_queue command_queue) {
 /// \return PI_SUCCESS
 pi_result cuda_piextQueueGetNativeHandle(pi_queue queue,
                                          pi_native_handle *nativeHandle) {
-  *nativeHandle = reinterpret_cast<pi_native_handle>(queue->get());
+  ScopedContext active(queue->get_context());
+  *nativeHandle = reinterpret_cast<pi_native_handle>(queue->get_compute());
```

Comment:
As mentioned in a previous comment, I suspect it would make sense to change the native handles for the CUDA backend. I am okay with leaving it as-is for now though.

@AerialMantis - What are your thoughts on this? Should the CUDA backend return the compute and transfer streams as native handle for the queue?

Reply:
Hi @steffenlarsen, apologies for the delay in replying I was on holiday the past week, we discussed this a fair bit as well, there could be a benefit to extending the backend interop such that `get_native_queue` returns a multiple streams, however, that would introduce the problem of how the user should interpret and use those streams which would complicate the more simple use case, so we decided to leave it as it is for now, and once we have some more user experience of it we should revisit it.

Reply:
Thank you, @AerialMantis! I think the primary problem I could see happen is the users extracting the CUstream and expecting that synchronizing on it would be the same as synchronizing on the SYCL queue, which would not necessarily be the case. Likewise, they could try and recreate the SYCL queue and expect it to act in the same way as the old one. I don't think it is necessarily an immediate problem, but it should be something to be considered while writing the backend documentation.